### PR TITLE
MC 1.8.3

### PR DESCRIPTION
**Pull request by DSH105**
Progress on the 1.8.3 update. Other related info can be found at #498.

Known issues (thanks to @mibby for reporting):
- [x] Pets do not move when spawned in (besides enderdragon).
- [x] Pets do not follow the owner, look at the owner, or teleport when a certain distance away.
- [x] '/pet list' causes an error. http://paste.ubuntu.com/10592262/
    - ***due to Spigot remapping the library.***
- [x] '/pet help' causes an error. http://paste.ubuntu.com/10592459/
    - ***due to Spigot remapping the library.***
- [x] Pet bat is not flying. It is spawned on the ground. As well as riding it, you are not able to fly with it.
    - **#512**
- [x] Pet blaze, same as bat. Can't fly. (Seems all mobs that would otherwise be flying aren't. Including wither, squid, ghast, etc.)
    - **#512**
- [x] All pets, can't jump while riding.
- [x] Pet enderdragon, controls are inverted. W = left, D = forward, S = right, A = forward, Can't change elevation, can't fly, can't use mouse movement, pet stays still while on it.
- [x] ~~Pet horse resets orientation (to where the horse is looking?) when right clicking on it.~~
    - ***Existing non-MC183 update issue. Perhaps even a non-EchoPet issue***
    - Marking as resolved as I'm not sure it's fixable.
- [x] ~~I personally keep human pet disabled, but enabling & testing, it says Human pet type is not compatible with this server version.~~
    - ***Intended: human pets are deprecated.***
- [x] ~~Pet names lose color value when toggling show/hide. Example, name a pet '&bBob'. It will show the Bob text in aqua color. Toggling hide and show, Bob is now white. Same occurs with relogging.~~
    - ***Existing non-MC183 update issue.***
    - For the sake of pushing this update out, I'll leave this one out for now. I know I've looked into it before and haven't located where to patch it yet.

New bugs:
- [ ] Guardians and rabbits slide along the ground as if they were snakes.
- [x] New permissions are missing.
- [x] New pets are missing in the pet selector.
- [x] Enderdragon flight does not function as expected while its owner is on the ground.

### CODE REVIEW

**Comment by mibby** [comment]
@DSH105 Commit seems to have fixed the pet tracking the player. Moving, looking, following, and teleporting when X amount of blocks away. However, many issues I reported are still present. Like flying, not being able to jump, etc. As well as if you teleport away from the pet, it does not teleport on you. Teleporting to coordinates far away from the pet with either /home, /spawn, or /tppos.

File: modules/v1_8_R2/src/main/java/com/dsh105/echopet/compat/nms/v1_8_R2/entity/EntityPet.java [hunk 1]
```diff
@@ -0,0 +1,531 @@
...
+        this.pet = pet;
+        this.initiateEntityPet();
+    }
+
+    private void initiateEntityPet() {
+        this.setSize();
+        this.fireProof = true;
+        if (this.FIELD_JUMP == null) {
+            try {
+                this.FIELD_JUMP = EntityLiving.class.getDeclaredField("aW");
```
Comment (DSH105) [inline modules/v1_8_R2/src/main/java/com/dsh105/echopet/compat/nms/v1_8_R2/entity/EntityPet.java]: this field needs to be checked against 1.8.3 mappings.

**Comment by DSH105** [comment]
@CaptainBern if you have any idea how to deal with the `/pet list`/`pet help` errors let me know :P

**Comment by DSH105** [comment]
@mibby dragon controls should now be correct

**Comment by mibby** [comment]
Great! :) I'll be ready to test any new commits for fixes. Did happen to see this in the logs though.

```
[10:51:14] [Server thread/WARN]: [EchoPet] Task #3702 for EchoPet v2.6.1-SNAPSHOT-b4 generated an exception
java.lang.NullPointerException
	at com.dsh105.echopet.api.pet.Pet$2.run(Pet.java:401) ~[?:?]
	at org.bukkit.craftbukkit.v1_8_R2.scheduler.CraftTask.run(CraftTask.java:71) ~[spigot.jar:git-Spigot-ea179b3-6e0120a]
	at org.bukkit.craftbukkit.v1_8_R2.scheduler.CraftScheduler.mainThreadHeartbeat(CraftScheduler.java:350) [spigot.jar:git-Spigot-ea179b3-6e0120a]
	at net.minecraft.server.v1_8_R2.MinecraftServer.A(MinecraftServer.java:718) [spigot.jar:git-Spigot-ea179b3-6e0120a]
	at net.minecraft.server.v1_8_R2.DedicatedServer.A(DedicatedServer.java:368) [spigot.jar:git-Spigot-ea179b3-6e0120a]
	at net.minecraft.server.v1_8_R2.MinecraftServer.z(MinecraftServer.java:651) [spigot.jar:git-Spigot-ea179b3-6e0120a]
	at net.minecraft.server.v1_8_R2.MinecraftServer.run(MinecraftServer.java:554) [spigot.jar:git-Spigot-ea179b3-6e0120a]
	at java.lang.Thread.run(Thread.java:745) [?:1.8.0_40]
```

**Comment by CaptainBern** [comment]
@DSH105: perhaps just shade in Gson too?

**Comment by DSH105** [comment]
@CaptainBern that's ugly and has the disadvantage of not sitting well with the Curse review crowd on DBO.

**Comment by CaptainBern** [comment]
@DSH105 I guess we could use another library instead then?

**Comment by DSH105** [comment]
I'll look into abstracting it or something I guess.

**Comment by South-Paw** [comment]
"Pet names lose color value" - Also occurs when using portals.

Could we also possibly get the new mobs added as pets too? (Bunnies, guardians (elder variant too) and endermites.

**Comment by drew5494** [comment]
Are developer builds available that has fixed the issue of pets not following?

**Comment by DSH105** [comment]
My CI builds this branch automatically (GitHub links it below), so any new commits listed above will have a corresponding dev build.

**Comment by drew5494** [comment]
Weird cause I am using that build yet pets don't follow me.
Hope all is well in your life @DSH105 you seemed a little down in the first thread. 
Remember that God has a plan for everyone.

**Comment by mibby** [comment]
Pets seem to follow me just fine with the latest 1.8.3 dev build. Make sure you are downloading the correct branch compile. 

https://ci.dsh105.com/job/EchoPet-v2%20(1.8.3)/

**Comment by DSH105** [comment]
@South-Paw we'll see, depending on how complex the entity's are and how many type variants I end up adding support for.

**Comment by DSH105** [comment]
Jumping should now be fine.
@mibby

**Comment by andrewkm** [comment]
Many thanks for continuing to support this plugin.
What is your donation email :) ?

**Comment by CaptainBern** [comment]
@andrewkm In the top right corner http://dev.bukkit.org/bukkit-plugins/echopet/ ;p

**Comment by andrewkm** [comment]
Ok sent a bit :) Thanks for your work, will definitely send more once I test and ensure all works well <3

**Comment by DSH105** [comment]
Thanks @andrewkm <3

**Comment by mibby** [comment]
Sweet, thanks DSH. :) I'll be keeping an eye open for more changes / fixes that need testing.

Another issue that was brought up to me by a player. With a pet enderdragon, when logging in and having it automatically spawn with the player, the pet's movement is glitchy / twitchy. Hiding and reshowing the pet resolves the issue.

**Comment by DSH105** [comment]
That's odd. EnderDragon behaviour shouldn't alter depending on when it's spawned. Is this only observable from the owner's perspective?

**Comment by mibby** [comment]
I believe so. I'll need to run some tests to confirm myself. Only relaying information of what others have experienced. :)

**Comment by andrewkm** [comment]
/pet help does not work
http://pastie.org/pastes/10045822/text

**Comment by DSH105** [comment]
@andrewkm yep, listed in the top post for me to fix :).

**Comment by xAphex** [comment]
is there a potential release date for EchoPet 1.8.3?
(like an official update with no bugs)

and thanks for continuing the support for EchoPet, everyone in my server is mad because EchoPet 2.6.0 won't work!

-Gabriel

**Comment by CaptainBern** [comment]
@xAphex As you see, this pull-request is going to be that "official update". :stuck_out_tongue_winking_eye:

**Comment by xAphex** [comment]
@CaptainBern Oh, ok, good to know!
can't wait for the official EchoPet 1.8.3 update!

**Comment by CILI0** [comment]
Hope the fix with human pet !

**Comment by DSH105** [comment]
Human pets are unstable and deprecated (unless they make a reappearance in EchoPet 3).

**Comment by whoshenry** [comment]
I am greatly looking forward to the release for EchoPet 1.8.3 or even EchoPet 3. Can I suggest you release Echopet as a premium plugin on Spigot, I am sure that this will get you motivated to continue regular development and people will be happy to pay for the plugin.

**Comment by xAphex** [comment]
@DSH105 you can keep human pets off for now, the thing the people care about is the pets!

and i'm so hyped about EchoPet 3, will look forward for that!

**Comment by South-Paw** [comment]
@whoshenry That'd be a real shame if @DSH105 did that, as a small community server we wouldn't be paying for it and therefore would just find another plugin... we only have enough money to host - not buy plugins for people :disappointed:

**Comment by Favorlock** [comment]
@South-Paw its not like he couldn't do what drtshock does with FactionsUUID, provide the precompiled jar on spigot for $3 or $5 and those who can't afford it can spend a little time learning how to compile a maven project or download from the CI.

Let me know if there is anything I can do to speed up the process @DSH105

**Comment by andrewkm** [comment]
Should we be using: https://ci.dsh105.com/job/EchoPet-v2/
Or still the 1.8.3 branch: https://ci.dsh105.com/job/EchoPet-v2%20%281.8.3%29/

Just curious as I haven't been keeping up sorry insanely busy.

**Comment by DSH105** [comment]
@andrewkm if you want 1.8.3 support, use the builds from the 'EchoPet-v2 1.8.3' job.

**Comment by xAphex** [comment]
@DSH105 Is there any bugs with the latest 1.8.3 dev build?

**Comment by DSH105** [comment]
Those new pets in the commits above are not tested *at all*. I haven't even fired up a test server myself to see if they load yet.

@xAphex see the top post/comment for a bug list.

**Comment by DSH105** [comment]
https://github.com/DSH105/PowerMessage/commit/cba55fa322e820e97f7f6fb69d87a505ece975d3 => [PowerMessage-b15](https://ci.dsh105.com/job/PowerMessage/15/) => [EchoPet-v2 (1.8.3)-b8](https://ci.dsh105.com/job/EchoPet-v2%20(1.8.3)/8/).

That should resolve the conflict experienced with the `/pet list` and `/pet help` commands.

File: modules/v1_8_R2/src/main/java/com/dsh105/echopet/compat/nms/v1_8_R2/entity/type/EntityRabbitPet.java [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+
+import com.dsh105.echopet.compat.api.entity.IPet;
+import com.dsh105.echopet.compat.api.entity.SizeCategory;
+import com.dsh105.echopet.compat.api.entity.type.nms.IEntityRabbitPet;
+import com.dsh105.echopet.compat.nms.v1_8_R2.entity.EntityAgeablePet;
+import com.dsh105.echopet.compat.nms.v1_8_R2.entity.EntityPet;
+import net.minecraft.server.v1_8_R2.World;
+import org.bukkit.entity.Rabbit;
+
+// TODO: hopping
```
Comment (DSH105) [inline modules/v1_8_R2/src/main/java/com/dsh105/echopet/compat/nms/v1_8_R2/entity/type/EntityRabbitPet.java]: existing rabbit behaviour might be suitable without hopping (needs testing).

**Comment by DSH105** [comment]
All previously reported issues are now fixed, except for:
- bats/blazes are not flying correctly (@mibby not sure exactly what you mean by this).
- rabbits 'slide' along the ground instead of walking.

**Comment by mibby** [comment]
@DSH105 They were being spawned on the ground "walking" instead of flying in the air following the user.

**Comment by DSH105** [comment]
I thought they always followed on the ground unless the player was flying.

**Comment by andrewkm** [comment]
/pet help is still not fixed.
Noticed you said all issues fixed, thought I'd report.

**Comment by DSH105** [comment]
I've tested it with b10 and it appears to be functional.

**Comment by andrewkm** [comment]
Can confirm still broken on B10. - /pet help

**Comment by andrewkm** [comment]
http://pastie.org/pastes/10071304/text?key=o9xjgjhogccte4cgbzy4ww

**Comment by DSH105** [comment]
I'll trigger a new build. Perhaps the dependencies weren't shaded in correctly.

**Comment by andrewkm** [comment]
Build 11 still broken.

**Comment by xLRDx** [comment]
Can confirm Build #11 is broken still. When doing /pet help.

**Comment by andrewkm** [comment]
Small bug:
https://github.com/DSH105/EchoPet/pull/510

**Comment by andrewkm** [comment]
Also looks like all your new pet permissions (rabbit and so forth) are not yet in the plugin.yml
Do you have them listed anywhere?

**Comment by andrewkm** [comment]
Another bug I am having on build 11:
http://www.ecocitycraft.com/forum/threads/bug-report-pet-name-colours-resetting.121683/

Information from above thread below:

Brief Description: I managed to reset my pet's colours.

Instructions: Give the pet a colourful name (I used 3hundred14pi with the same colours as 3point's ingame nick). Use /fly to move about 200 blocks away from the pet that is still waiting on the ground. (I flew from /Warp greenmarket2 to /Warp greenmarket1 to test). Return to the starting location, the pet is gone, it will not be immediately teleported when you hit the ground as it normally should. /pet call brings the pet to your location, but it resets all colour codes.

How many times did you recreate this bug?: 2 times out of 2 tries.

Result: A pet with a white nickname.

Expected Result: A pet with a coloured nickname.

Evidence: I can upload screenshots or a video if needed. However, seems to be easily reproducable.

**Comment by DSH105** [comment]
@andrewkm above commit deals with any missing permissions. I *think* the config should automatically generate the missing 'pet selector' slots for the new pet types on its own.

**Comment by andrewkm** [comment]
I have just tested build 14 and generated a new config. Can confirm it does *not* generate the new slots.

**Comment by DSH105** [comment]
Alright, I'll send out a search party of monkeys to find where that stuff is generated.

**Comment by andrewkm** [comment]
Affirmative, I will inform PETA animal rights association that you treat said monkeys with proper care so as not to land you in any trouble.

**Comment by DSH105** [comment]
@xLRDx :+1: :smile: 

@andrewkm Name colour bug has been/is listed above.

**Comment by mibby** [comment]
Pet help still seems to be broken on the newest dev build.

```
[21:06:34] [Server thread/INFO]: mibby issued server command: /pet help
[21:06:34] [Server thread/WARN]: java.lang.ClassNotFoundException: net.minecraft.server.v1_8_R2.ChatSerializer
[21:06:34] [Server thread/WARN]: 	at org.bukkit.plugin.java.PluginClassLoader.findClass(PluginClassLoader.java:91)
[21:06:34] [Server thread/WARN]: 	at org.bukkit.plugin.java.PluginClassLoader.findClass(PluginClassLoader.java:86)
[21:06:34] [Server thread/WARN]: 	at java.lang.ClassLoader.loadClass(ClassLoader.java:424)
[21:06:34] [Server thread/WARN]: 	at java.lang.ClassLoader.loadClass(ClassLoader.java:357)
[21:06:34] [Server thread/WARN]: 	at java.lang.Class.forName0(Native Method)
[21:06:34] [Server thread/WARN]: 	at java.lang.Class.forName(Class.java:264)
[21:06:34] [Server thread/WARN]: 	at com.dsh105.echopet.libs.dsh105.reflection.Reflection.getClass(Reflection.java:64)
[21:06:34] [Server thread/WARN]: 	at com.dsh105.echopet.libs.dsh105.reflection.Reflection.getNMSClass(Reflection.java:83)
[21:06:34] [Server thread/WARN]: 	at com.dsh105.echopet.libs.dsh105.core.PowerMessage.<clinit>(PowerMessage.java:66)
[21:06:34] [Server thread/WARN]: 	at com.dsh105.echopet.commands.HelpEntry.getPowerMessage(HelpEntry.java:95)
[21:06:34] [Server thread/WARN]: 	at com.dsh105.echopet.commands.PetCommand.getHelp(PetCommand.java:55)
[21:06:34] [Server thread/WARN]: 	at com.dsh105.echopet.commands.PetCommand.onCommand(PetCommand.java:289)
[21:06:34] [Server thread/WARN]: 	at com.dsh105.echopet.commands.util.DynamicPluginCommand.execute(DynamicPluginCommand.java:46)
[21:06:34] [Server thread/WARN]: 	at org.bukkit.command.SimpleCommandMap.dispatch(SimpleCommandMap.java:141)
[21:06:34] [Server thread/WARN]: 	at org.bukkit.craftbukkit.v1_8_R2.CraftServer.dispatchCommand(CraftServer.java:646)
[21:06:34] [Server thread/WARN]: 	at net.minecraft.server.v1_8_R2.PlayerConnection.handleCommand(PlayerConnection.java:1137)
[21:06:34] [Server thread/WARN]: 	at net.minecraft.server.v1_8_R2.PlayerConnection.a(PlayerConnection.java:972)
[21:06:34] [Server thread/WARN]: 	at net.minecraft.server.v1_8_R2.PacketPlayInChat.a(PacketPlayInChat.java:45)
[21:06:34] [Server thread/WARN]: 	at net.minecraft.server.v1_8_R2.PacketPlayInChat.a(PacketPlayInChat.java:1)
[21:06:34] [Server thread/WARN]: 	at net.minecraft.server.v1_8_R2.PlayerConnectionUtils$1.run(SourceFile:13)
[21:06:34] [Server thread/WARN]: 	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
[21:06:34] [Server thread/WARN]: 	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
[21:06:34] [Server thread/WARN]: 	at net.minecraft.server.v1_8_R2.SystemUtils.a(SourceFile:60)
[21:06:34] [Server thread/WARN]: 	at net.minecraft.server.v1_8_R2.MinecraftServer.A(MinecraftServer.java:710)
[21:06:34] [Server thread/WARN]: 	at net.minecraft.server.v1_8_R2.DedicatedServer.A(DedicatedServer.java:368)
[21:06:34] [Server thread/WARN]: 	at net.minecraft.server.v1_8_R2.MinecraftServer.z(MinecraftServer.java:651)
[21:06:34] [Server thread/WARN]: 	at net.minecraft.server.v1_8_R2.MinecraftServer.run(MinecraftServer.java:554)
[21:06:34] [Server thread/WARN]: 	at java.lang.Thread.run(Thread.java:745)
[21:06:34] [Server thread/ERROR]: null
org.bukkit.command.CommandException: Unhandled exception executing 'pet help' in com.dsh105.echopet.commands.util.DynamicPluginCommand(pet)
	at org.bukkit.command.SimpleCommandMap.dispatch(SimpleCommandMap.java:148) ~[spigot.jar:git-Spigot-4c7d0c0-7d020a7]
	at org.bukkit.craftbukkit.v1_8_R2.CraftServer.dispatchCommand(CraftServer.java:646) ~[spigot.jar:git-Spigot-4c7d0c0-7d020a7]
	at net.minecraft.server.v1_8_R2.PlayerConnection.handleCommand(PlayerConnection.java:1137) [spigot.jar:git-Spigot-4c7d0c0-7d020a7]
	at net.minecraft.server.v1_8_R2.PlayerConnection.a(PlayerConnection.java:972) [spigot.jar:git-Spigot-4c7d0c0-7d020a7]
	at net.minecraft.server.v1_8_R2.PacketPlayInChat.a(PacketPlayInChat.java:45) [spigot.jar:git-Spigot-4c7d0c0-7d020a7]
	at net.minecraft.server.v1_8_R2.PacketPlayInChat.a(PacketPlayInChat.java:1) [spigot.jar:git-Spigot-4c7d0c0-7d020a7]
	at net.minecraft.server.v1_8_R2.PlayerConnectionUtils$1.run(SourceFile:13) [spigot.jar:git-Spigot-4c7d0c0-7d020a7]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511) [?:1.8.0_40]
	at java.util.concurrent.FutureTask.run(FutureTask.java:266) [?:1.8.0_40]
	at net.minecraft.server.v1_8_R2.SystemUtils.a(SourceFile:60) [spigot.jar:git-Spigot-4c7d0c0-7d020a7]
	at net.minecraft.server.v1_8_R2.MinecraftServer.A(MinecraftServer.java:710) [spigot.jar:git-Spigot-4c7d0c0-7d020a7]
	at net.minecraft.server.v1_8_R2.DedicatedServer.A(DedicatedServer.java:368) [spigot.jar:git-Spigot-4c7d0c0-7d020a7]
	at net.minecraft.server.v1_8_R2.MinecraftServer.z(MinecraftServer.java:651) [spigot.jar:git-Spigot-4c7d0c0-7d020a7]
	at net.minecraft.server.v1_8_R2.MinecraftServer.run(MinecraftServer.java:554) [spigot.jar:git-Spigot-4c7d0c0-7d020a7]
	at java.lang.Thread.run(Thread.java:745) [?:1.8.0_40]
Caused by: java.lang.ExceptionInInitializerError
	at com.dsh105.echopet.commands.HelpEntry.getPowerMessage(HelpEntry.java:95) ~[?:?]
	at com.dsh105.echopet.commands.PetCommand.getHelp(PetCommand.java:55) ~[?:?]
	at com.dsh105.echopet.commands.PetCommand.onCommand(PetCommand.java:289) ~[?:?]
	at com.dsh105.echopet.commands.util.DynamicPluginCommand.execute(DynamicPluginCommand.java:46) ~[?:?]
	at org.bukkit.command.SimpleCommandMap.dispatch(SimpleCommandMap.java:141) ~[spigot.jar:git-Spigot-4c7d0c0-7d020a7]
	... 14 more
Caused by: java.lang.NullPointerException
	at com.dsh105.echopet.libs.dsh105.core.PowerMessage.<clinit>(PowerMessage.java:66) ~[?:?]
	at com.dsh105.echopet.commands.HelpEntry.getPowerMessage(HelpEntry.java:95) ~[?:?]
	at com.dsh105.echopet.commands.PetCommand.getHelp(PetCommand.java:55) ~[?:?]
	at com.dsh105.echopet.commands.PetCommand.onCommand(PetCommand.java:289) ~[?:?]
	at com.dsh105.echopet.commands.util.DynamicPluginCommand.execute(DynamicPluginCommand.java:46) ~[?:?]
	at org.bukkit.command.SimpleCommandMap.dispatch(SimpleCommandMap.java:141) ~[spigot.jar:git-Spigot-4c7d0c0-7d020a7]
	... 14 more
```

**Comment by DSH105** [comment]
Build 16 seems to have finally picked up on the upstream commits (https://github.com/DSH105/PowerMessage/commit/199787c99949008486681da720fa6cb69ab7c8f4) that were implemented to resolve the issues with `/pet help` and `/pet list`.

**Comment by DSH105** [comment]
@andrewkm config generation should now include those new pet types (I tested myself with b17 to confirm).

**Comment by andrewkm** [comment]
Looks like this bug:

http://www.ecocitycraft.com/forum/threads/bug-report-pet-name-colours-resetting.121683/

Has just got a bit more severe:

"@andrewkm, small update...before, the bug would only occur if flying away. Now it also happens with any form of teleportation within the world. If I do /spawn at the market, the pet will be hidden until I /pet call a de-coloured version. Teleports only work normally if traveling between worlds now... "

**Comment by DSH105** [comment]
It shouldn't have been exclusive to flying in the first place, as the teleportation was all executed the same way - #458.

**Comment by South-Paw** [comment]
Finally got around to putting it on our server for some testing and all is going well! Will let you know if any problems are encountered. Thanks DSH <3 <3 <3

(Used: https://ci.dsh105.com/job/EchoPet-v2%20(1.8.3)/)

**Comment by mibby** [comment]
@DSH105 Pet enderdragon seems to be partially broken. When on the ground and spawning in the pet, the enderdragon will constantly be glitching / teleporting to the player when it tries to fly away. When in the air flying and spawning in the pet, the enderdragon behavior is perfectly natural. It flies around.

**Comment by DSH105** [comment]
That's probably because the obfuscation is slightly out and I'm not entirely enthusiastic about digging painfully through that nms file.

**Comment by DSH105** [comment]
diff tools safe the day :+1:

**Comment by DSH105** [comment]
If anyone feels inclined to have a crack at fixing the rabbit, feel free.

**Comment by xAphex** [comment]
Mojang released Minecraft 1.8.4 today, will this mod get updated to 1.8.4?

**Comment by DSH105** [comment]
well considering this one isn't even merged yet.. :p

**Comment by andrewkm** [comment]
Working great on 1.8.4

**Comment by xLRDx** [comment]
So if i am correct in wondering all that is really left to fix on this is the rabbits and guardians?

**Comment by DSH105** [comment]
Correct.

**Comment by mibby** [comment]
Reminder to add the new pets to the pet selector. :)

**Comment by DSH105** [comment]
@mibby that was done a while ago https://github.com/DSH105/EchoPet/commit/d091e2b0748d7a10e8bad9075800a32c08c4389b.

**Comment by mibby** [comment]
@DSH105 Odd. I don't see the mobs with my pet selector.

**Comment by DSH105** [comment]
@mibby was your config regenerated?

**Comment by mibby** [comment]
Nope, guess that explains it. Thanks. :)

**Comment by DSH105** [comment]
(needs testing).

**Comment by xAphex** [comment]
i have tested the newest version (#46) on the latest spigot 1.8.3 build (Spigot-870264a-0a645a2) and there is no errors! everything is smooth
i have a question: are you gonna update EchoPet to Spigot 1.8.4? (it was released a week ago)

Thank you

**Comment by DSH105** [comment]
It was merged and released to DBO a week and s half ago.

**Comment by NightRiider** [comment]
Whats the link I should be downloading the latest version of EchoPet from?

**Comment by DSH105** [comment]
* [Latest DBO release](http://dev.bukkit.org/bukkit-plugins/echopet/files/34-echo-pet-v2-8-0/)
* [Latest dev build](https://ci.dsh105.com/job/EchoPet-v2/61/)

**Pull request merged**